modules/claims_api/app/controllers/concerns/claims_api/target_veteran.rb

```diff
@@ -33,21 +34,45 @@ def build_target_veteran(veteran_id:, loa:)
                     'Please submit an issue at ask.va.gov or call 1-800-MyVA411 (800-698-2411) for assistance.'
           )
         end
-        populate_target_veteran(mpi_profile_from(target_veteran), target_veteran)
-      end
 
-      def mpi_profile_from(target_veteran)
-        mpi_profile = target_veteran&.mpi&.mvi_response&.profile || {}
-        if mpi_profile[:participant_id].blank?
+        built_target_veteran = populate_target_veteran(mpi_profile_from(target_veteran), target_veteran)
+
+        # rubocop:disable Layout/LineLength
+        if built_target_veteran.mpi.participant_id.blank? && Flipper.enabled?(:lighthouse_claims_api_v2_add_person_proxy)
+          # rubocop:enable Layout/LineLength
           claims_logging('unable_to_locate_participant_id',
-                         message: 'unable_to_locate_participant_id on request in target veteran.')
+                         message: 'unable_to_locate_participant_id.')
 
-          raise ::Common::Exceptions::UnprocessableEntity.new(
-            detail: "Unable to locate Veteran's Participant ID in Master Person Index (MPI). " \
-                    'Please submit an issue at ask.va.gov or call 1-800-MyVA411 (800-698-2411) for assistance.'
-          )
+          built_target_veteran_response = built_target_veteran.mpi_no_cache.add_person_proxy
+
+          if built_target_veteran_response.ok?
+            built_target_veteran = populate_target_veteran(mpi_profile_from(target_veteran),
+                                                           target_veteran)
+          end
+
+          if built_target_veteran.mpi_no_cache.participant_id.blank?
+            raise ::Common::Exceptions::UnprocessableEntity.new(detail:
+              "Unable to locate Veteran's Participant ID in Master Person Index (MPI). " \
+              'Please submit an issue at ask.va.gov or call 1-800-MyVA411 (800-698-2411) ' \
+              'for assistance.')
+          end
         end
-        mpi_profile
+
+        built_target_veteran
+      end
+      # rubocop:enable Metrics/MethodLength
+
+      def mpi_profile_from(target_veteran)
+        target_veteran&.mpi&.mvi_response&.profile || {}
+        # if mpi_profile[:participant_id].blank?
+        #   claims_logging('unable_to_locate_participant_id',
+        #                  message: 'unable_to_locate_participant_id on request in target veteran.')
+
+        #   raise ::Common::Exceptions::UnprocessableEntity.new(
+        #     detail: "Unable to locate Veteran's Participant ID in Master Person Index (MPI). " \
+        #             'Please submit an issue at ask.va.gov or call 1-800-MyVA411 (800-698-2411) for assistance.'
+        #   )
+        # end
```

Comment:
This only appears to be getting called from in here, so the check needs to be removed because we want to call `add_person_prooxy` in the even this returns and object with no `participant_id`.